pkg/frontend/querymiddleware/querysharding_test.go

```diff
@@ -182,6 +182,18 @@ func TestQuerySharding_Correctness(t *testing.T) {
 			query:                  `sum by(unique) (rate(metric_counter{group_1="0"}[1m]))`,
 			expectedShardedQueries: 1,
 		},
+		`group by (group_1) (metric_counter)`: {
+			query:                  `group by (group_1) (metric_counter)`,
+			expectedShardedQueries: 1,
+		},
+		`group by (group1) (group by (group_1, group_2) (metric_counter))`: {
+			query:                  `group by (group1) (group by (group_1, group_2) (metric_counter))`,
+			expectedShardedQueries: 1,
+		},
+		`count by (group1) (group by (group_1, group_2) (metric_counter))`: {
+			query:                  `count by (group1) (group by (group_1, group_2) (metric_counter))`,
+			expectedShardedQueries: 1,
+		},
```

Comment:
group1 -> group_1?

Reply:
Thanks, good catch. Fixed.

Reply:
unfortunately, that's as much as I can review this PR 😢 The rest SGTM, but i don't feel confident stamping an approval